packages/react/src/components/Button/SplitButton.tsx

```diff
@@ -0,0 +1,155 @@
+  button?: ShorthandValue<ButtonProps>
+
+  /** Element to be rendered in-place where the popup is defined. */
+  menu?: ShorthandValue<MenuProps>
+
+  /**
+   * Called after user's click.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onClick?: ComponentEventHandler<ButtonProps>
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  primary?: boolean
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  secondary?: boolean
+}
+
+class SplitButton extends UIComponent<WithAsProp<SplitButtonProps>> {
+  static create: Function
+
+  static displayName = 'SplitButton'
+
+  static className = 'ui-splitbutton'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    button: customPropTypes.itemShorthand,
+    menu: customPropTypes.itemShorthand,
+    onClick: PropTypes.func,
+    onToggle: PropTypes.func,
+    primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
+    secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
+  }
+
+  static defaultProps = {
+    accessibility: splitButtonBehavior,
+    as: 'span',
+    primary: true,
+  }
+
+  buttonRef = React.createRef<HTMLElement>()
+  menuButtonRef = React.createRef<MenuButton>()
+
+  actionHandlers = {
+    closeMenuAndFocusButton: () => {
+      this.closeMenu()
+      this.buttonRef.current.focus()
+    },
+    openAndFocusFirst: e => this.openAndFocusFirst(e),
+  }
+
+  closeMenu() {
+    this.menuButtonRef.current.closeMenu()
+  }
+
+  openAndFocusFirst(e) {
+    this.menuButtonRef.current.openAndFocus(e, 'first')
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    variables,
+    styles,
+    unhandledProps,
+  }): React.ReactNode {
+    const { button, menu, primary, secondary } = this.props
+
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...unhandledProps}
+        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
+      >
+        <Ref innerRef={this.buttonRef}>
+          {Button.create(button, {
+            defaultProps: {
+              styles: styles.button,
+              variables: variables.button,
+              primary,
+              secondary,
+              ...accessibility.attributes.button,
+              ...applyAccessibilityKeyHandlers(accessibility.keyHandlers.button, button),
+            },
+            overrideProps: (predefinedProps?: ButtonProps) => ({
+              onClick: (e: React.SyntheticEvent, buttonProps: ButtonProps) => {
+                _.invoke(predefinedProps, 'onClick', e, buttonProps)
+                _.invoke(this.props, 'onClick', e, buttonProps)
+              },
+            }),
+          })}
+        </Ref>
+        <MenuButton
```

Comment:
Please use factories, i.e. `MenuButton.create()`

Reply:
But we don't have shorthand for `menuButton` yet.